### PR TITLE
chore(www): Refactor `ecosystem-section`

### PR DESCRIPTION
**Pull request by fk**
Switch from template literal/string styles to [`sx`](https://theme-ui.com/sx-prop) and object styles.

### CODE REVIEW

**Comment by fk** [comment]
I think this should be good now, too.
Will see to get the last of those PRs fixed up before I gotta run.

https://www.youtube.com/watch?v=cNplZrRSjeI

File: www/src/components/ecosystem/ecosystem-section.js [hunk 1]
```diff
@@ -105,14 +99,14 @@ const EcosystemSection = ({
   featuredItems,
   className,
 }) => (
-  <EcosystemSectionRoot className={className}>
+  <section sx={ecosystemSectionRootStyles} className={className}>
```
Comment (tesseralis) [inline www/src/components/ecosystem/ecosystem-section.js:102]: I don't think the `className` prop is used anywhere? Can probably just delete.

File: www/src/components/ecosystem/ecosystem-section.js [hunk 2]
```diff
@@ -1,100 +1,94 @@
...
-  font-size: ${p => p.theme.fontSizes[1]};
-  font-weight: normal;
-  letter-spacing: ${p => p.theme.letterSpacings.tracked};
-  margin: 0;
-  margin-top: ${p => p.theme.space[5]};
-  text-transform: uppercase;
-`
+const iconStyles = {
+  display: `block`,
+  height: t => t.space[7],
```
Comment (tesseralis) [inline www/src/components/ecosystem/ecosystem-section.js:65]: theme-ui has a `sizes` property that you can define if you want to add to `gatsby-design-tokens.` Until then maybe it's better to keep these as explicit px/rem values?

File: www/src/components/ecosystem/ecosystem-section.js [hunk 3]
```diff
@@ -1,100 +1,94 @@
...
+  a: {
+    textDecoration: `none`,
+  },
+}
 
-  span {
-    margin: 0 ${p => p.theme.space[1]} 0 0;
-  }
-`
+export const Header = props => (
```
[the remark this answers is not in the document]
Comment (fk) [reply www/src/components/ecosystem/ecosystem-section.js:43]: Will check where else this is used (if).

File: www/src/components/ecosystem/ecosystem-section.js (hunk 2 above)
Comment (fk) [reply www/src/components/ecosystem/ecosystem-section.js:64]: Yeah, would have been best. Establishing a relation to `space` was too tempting 😅

File: www/src/components/ecosystem/ecosystem-section.js (hunk 1 above)
Comment (fk) [reply www/src/components/ecosystem/ecosystem-section.js:101]: 👍

Comment (fk) [reply www/src/components/ecosystem/ecosystem-section.js:101]: Yep, but need `{...rest}` still for https://github.com/gatsbyjs/gatsby/blob/c1368c06fb975bd792ebb8f9d0c5a5e4ebcba388/www/src/components/homepage/homepage-ecosystem.js#L30-L45

Comment (fk) [reply www/src/components/ecosystem/ecosystem-section.js:101]: Btw. we lost those icons on the "Starters" and "Plugins" cards I think?

image

File: www/src/components/ecosystem/ecosystem-section.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (fk) [reply www/src/components/ecosystem/ecosystem-section.js]: ✅

File: www/src/components/ecosystem/ecosystem-section.js [hunk 4]
```diff
@@ -103,16 +94,16 @@ const EcosystemSection = ({
   icon,
   links,
   featuredItems,
-  className,
+  ...rest
```
Comment (tesseralis) [inline www/src/components/ecosystem/ecosystem-section.js:97]: Do we need this `…rest` here?

**Pull request merged**